executable-spec/src/Cardano/Ledger/Spec/STS/Update/Ideation.hs

```diff
@@ -51,31 +55,42 @@ instance STS IDEATION where
     | NoSIPToReveal Data.SIP
     | SIPAlreadyRevealed Data.SIP
     | InvalidAuthor Core.VKey
+    | SIPFailedToBeRevealed Data.SIP
     deriving (Eq, Show)
 
   initialRules = [ pure $! mempty ]
 
   transitionRules = [
     do
       TRC ( participants
-          , st@Data.State { submittedSIPs, revealedSIPs }
+          , st@Data.State { commitedSIPs, submittedSIPs, revealedSIPs }
           , sig
           ) <- judgmentContext
       case sig of
-        Submit sip -> do
+        Submit sipc sip -> do
           author sip ∈ dom participants ?! InvalidAuthor (author sip)
           sip ∉ submittedSIPs ?! SIPAlreadySubmitted sip
-          pure $! st { submittedSIPs = Set.insert sip submittedSIPs }
+          pure $! st { commitedSIPs = Map.insert (Data._commit sipc) (sipc) commitedSIPs
+                     , submittedSIPs = Set.insert sip submittedSIPs
+                     }
         Reveal sip -> do
           author sip ∈ dom participants ?! InvalidAuthor (author sip)
           sip ∈ submittedSIPs ?! NoSIPToReveal sip
           sip ∉ revealedSIPs ?! SIPAlreadyRevealed sip
+
+          -- debug
+          -- let !dummy3 = Debug.trace ("calcCommit: " ++ show (Data.calcCommit sip)) True
+          --     !dummy4 = Debug.trace ("LookUpCommit: " ++ show (Map.lookup (Data.calcCommit sip) commitedSIPs)) True
+
+          case Map.lookup (Data.calcCommit sip) commitedSIPs of
```

Comment:
Using the `Relation` operators I think we could simplify this to `calcCommit sip ∈ dom commitedSIPs`. Slightly more inefficient but closer to a formal notation. Otherwise we can just use [`member`](https://www.stackage.org/haddock/lts-13.29/containers-0.6.0.1/Data-Map-Strict.html#v:member)

Reply:
OK I will change it.

Reply:
I have changed it